### PR TITLE
chore(tests): e2e tests

### PR DESCRIPTION
**Pull request by sjelin**
Travis only runs a subset of tests via `npm run test_travis`, but `npm test` runs a full suite.
See angular#165 for details on why travis cannot run all tests.

### CODE REVIEW

**Comment by sjelin** [comment]
tavis failing :slightly_frowning_face:

**Comment by sjelin** [comment]
I only ended up being able to support a subset of tests on travis, but this should be good to go once we merge in https://github.com/angular/webdriver-manager/pull/162

**Comment by sjelin** [comment]
Unblocked!  We should still merge https://github.com/angular/webdriver-manager/pull/161 first though

File: gulpfile.js [hunk 1]
```diff
@@ -45,18 +48,57 @@ gulp.task('format', () => {
...
-gulp.task('test', ['format', 'build'], function(done) {
-  var opt_arg = [];
-  opt_arg.push('node_modules/jasmine/bin/jasmine.js');
-  runSpawn(done, process.execPath, opt_arg);
+// Command line commands
+gulp.task('update', ['build'], function(done) {
+  runSpawn(process.execPath, ['bin/webdriver-manager', 'update', '--android',
+      '--android-accept-licenses'], done);
 });
+gulp.task('start', ['build', 'shutdown'], function(done) {
```
Comment (mgiambalvo) [inline gulpfile.js]: fyi, Craig and I have talked about moving away from gulp to just shell scripts. That's similar to what Angular does for their CI. I think it's a bit easier to understand, when all you're doing is spawning a bunch of processes in sequence.

File: gulpfile.js [hunk 2]
```diff
@@ -45,18 +48,57 @@ gulp.task('format', () => {
...
+  runSequence('update', 'test:e2e:no_update', done);
+});
+gulp.task('test:e2e:no_screen', function(done) {
+  runSequence('update', 'start:no_screen', 'test:e2e:inner:no_screen', 'shutdown', done);
+});
+
+
+gulp.task('test', ['format', 'test:unit', 'test:e2e']);
+gulp.task('test:no_update', ['format', 'test:unit', 'test:e2e:no_update']);
+gulp.task('test:no_screen', ['format', 'test:unit', 'test:e2e:no_screen']);
```
Comment (mgiambalvo) [inline gulpfile.js]: I think 'headless' is the more common term for this kind of thing than 'no_screen', but it's up to you.

**Comment by mgiambalvo** [comment]
lgtm! Just some optional suggestions

File: gulpfile.js (hunk 1 above)
Comment (sjelin) [inline gulpfile.js]: I'll leave the transition to Craig.  But I do agree.

File: gulpfile.js (hunk 2 above)
Comment (sjelin) [inline gulpfile.js]: +1 much better name

**Pull request merged**